sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilderTest.java

```diff
@@ -0,0 +1,108 @@
+        final String subscriptionKey = getSubscriptionKey();
+        Objects.requireNonNull(subscriptionKey, "`AZURE_TEXT_ANALYTICS_SUBSCRIPTION_KEY` expected to be set.");
+
+        final TextAnalyticsSubscriptionKeyCredential credential =
+            new TextAnalyticsSubscriptionKeyCredential(subscriptionKey);
+
+        final TextAnalyticsClientBuilder clientBuilder = new TextAnalyticsClientBuilder()
+            .subscriptionKey(credential)
+            .endpoint(endpoint);
+
+        if (interceptorManager.isPlaybackMode()) {
+            clientBuilder.httpClient(interceptorManager.getPlaybackClient());
+        } else {
+            clientBuilder.httpClient(new NettyAsyncHttpClientBuilder().wiretap(true).build())
+                .addPolicy(interceptorManager.getRecordPolicy());
+        }
+
+        final TextAnalyticsAsyncClient client = clientBuilder.buildAsyncClient();
+
+        // Action and Assert
+        StepVerifier.create(client.detectLanguage("This is a test English Text"))
+            .assertNext(response -> validateDetectedLanguages(
+                Arrays.asList(new DetectedLanguage("English", "en", 1.0)),
+                response.getDetectedLanguages()))
+            .verifyComplete();
+
+        // Update to invalid subscription key
+        credential.updateCredential("invalid key");
+
+        // Action and Assert
+        StepVerifier.create(client.detectLanguage("This is a test English Text"))
+            .verifyError(HttpResponseException.class);
+    }
+
+    @Test
+    public void rotateSharedKeyCredentialSyncClient() {
+        // Arrange
+        final String endpoint = getEndpoint();
+        Objects.requireNonNull(endpoint, "`endpoint` expected to be set.");
+        final String subscriptionKey = getSubscriptionKey();
+        Objects.requireNonNull(subscriptionKey, "`AZURE_TEXT_ANALYTICS_SUBSCRIPTION_KEY` expected to be set.");
+
+        final TextAnalyticsSubscriptionKeyCredential credential =
+            new TextAnalyticsSubscriptionKeyCredential(subscriptionKey);
+
+        final TextAnalyticsClientBuilder clientBuilder = new TextAnalyticsClientBuilder()
+            .subscriptionKey(credential)
+            .endpoint(endpoint);
+
+        if (interceptorManager.isPlaybackMode()) {
+            clientBuilder.httpClient(interceptorManager.getPlaybackClient());
+        } else {
+            clientBuilder.httpClient(new NettyAsyncHttpClientBuilder().wiretap(true).build())
+                .addPolicy(interceptorManager.getRecordPolicy());
+        }
+
+        final TextAnalyticsClient client = clientBuilder.buildClient();
+
+        // Action and Assert
+        validateDetectedLanguages(Arrays.asList(new DetectedLanguage("English", "en", 1.0)),
+            client.detectLanguage("This is a test English Text").getDetectedLanguages());
+
+        // Update to invalid subscription key
+        credential.updateCredential("invalid key");
+
```

Comment:
What happens when client update with valid key and after that all the calls should be successful. Is there a test for that ?